src/App.php

```diff
@@ -311,16 +313,242 @@ public function __construct($basepath)
+	 *
+	 * First loads all activated addons default configuration throught the load_config hook, then load the local.ini.php
+	 * again to overwrite potential local addon configuration.
+	 */
+	private function loadAddonConfig()
+	{
+		// Loads addons default config
+		Core\Addon::callHooks('load_config');
+
+		// Load the local addon config file to overwritten default addon config values
+		if (file_exists($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'addon.ini.php')) {
+			$this->loadConfigFile($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'addon.ini.php');
+		}
+	}
+
+	/**
+	 * Loads the default timezone
+	 *
+	 * Include support for legacy $default_timezone
+	 *
+	 * @global string $default_timezone
+	 */
+	private function loadDefaultTimezone()
+	{
+		if ($this->getConfigValue('system', 'default_timezone')) {
+			$this->timezone = $this->getConfigValue('system', 'default_timezone');
+		} else {
+			global $default_timezone;
+			$this->timezone = !empty($default_timezone) ? $default_timezone : 'UTC';
+		}
+
+		if ($this->timezone) {
+			date_default_timezone_set($this->timezone);
+		}
+	}
+
+	/**
+	 * Figure out if we are running at the top of a domain or in a sub-directory and adjust accordingly
+	 */
+	private function determineUrlPath()
+	{
+		$this->urlpath = $this->getConfigValue('system', 'urlpath');
 
-		/**
-		 * Load the configuration file which contains our DB credentials.
-		 * Ignore errors. If the file doesn't exist or is empty, we are running in
-		 * installation mode.	 *
+		/* SCRIPT_URL gives /path/to/friendica/module/parameter
+		 * QUERY_STRING gives pagename=module/parameter
+		 *
+		 * To get /path/to/friendica we perform dirname() for as many levels as there are slashes in the QUERY_STRING
 		 */
-		$this->mode = ((file_exists('.htconfig.php') && filesize('.htconfig.php')) ? App::MODE_NORMAL : App::MODE_INSTALL);
+		if (!empty($_SERVER['SCRIPT_URL'])) {
+			// Module
+			if (!empty($_SERVER['QUERY_STRING'])) {
+				$path = trim(dirname($_SERVER['SCRIPT_URL'], substr_count(trim($_SERVER['QUERY_STRING'], '/'), '/') + 1), '/');
+			} else {
+				// Root page
+				$path = trim($_SERVER['SCRIPT_URL'], '/');
+			}
+
+			if ($path && $path != $this->urlpath) {
+				$this->urlpath = $path;
+			}
+		}
+	}
+
+	/**
+	 * Sets the App mode
+	 *
+	 * - App::MODE_INSTALL    : Either the database connection can't be established or the config table doesn't exist
+	 * - App::MODE_MAINTENANCE: The maintenance mode has been set
+	 * - App::MODE_NORMAL     : Normal run with all features enabled
+	 *
+	 * @return type
+	 */
+	private function determineMode()
+	{
+		$this->mode = 0;
+
+		if (!file_exists($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'local.ini.php')
+			&& !file_exists($this->basepath . DIRECTORY_SEPARATOR . '.htconfig.php')) {
+			return;
+		}
+
+		$this->mode |= App::MODE_LOCALCONFIGPRESENT;
+
+		if (!\dba::connected()) {
```

Comment:
Is there a reason for the ```\``` here?

Reply:
Yes, the `App` class is in the `Friendica` namespace, which means that all non-fully formed class names (without the leading `\`) will be looked up under the `Friendica` namespace. It is the same issue with `Exception` for example. So far we just didn't use `dba` in this file.

Reply:
Maybe then`use dba`; in header instead?

Reply:
My next task will be to move the `dba` class to `src/`, so it's going to change.